File: internal/parser/testdata/sample.c
#include <stdio.h>
#include "config.h"

typedef struct {
    char* host;
    int port;
} Config;

struct Server {
    Config config;
    int running;
};

enum Status {
    ACTIVE,
    INACTIVE
};

Config* create_config(const char* host, int port);

void run_server(Config* config) {
    printf("%s:%d\n", config->host, config->port);
}

File: internal/parser/testdata/sample.cs
using System;
using System.Collections.Generic;

namespace Server
{
    public struct Point
    {
        public double X;
        public double Y;
    }

    public interface IHandler
    {
        void Handle(string request);
    }

    public class Config
    {
        private string _host;
        private int _port;

        public Config(string host, int port)
        {
            _host = host;
            _port = port;
        }

        public string Address()
        {
            return $"{_host}:{_port}";
        }
    }

    public enum Status
    {
        Active,
        Inactive
    }
}
